src/Opulence/Framework/Databases/Console/Commands/FixMigrationsCommand.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * Opulence
+ *
+ * @link      https://www.opulencephp.com
+ * @copyright Copyright (C) 2017 David Young
+ * @license   https://github.com/opulencephp/Opulence/blob/master/LICENSE.md
+ */
+
+namespace Opulence\Framework\Databases\Console\Commands;
+
+use Exception;
+use Opulence\Console\Commands\Command;
+use Opulence\Console\Responses\IResponse;
+use Opulence\Databases\IConnection;
+use Opulence\QueryBuilders\MySql\QueryBuilder as MySqlQueryBuilder;
+use Opulence\QueryBuilders\PostgreSql\QueryBuilder as PostgreSqlQueryBuilder;
+use Opulence\QueryBuilders\QueryBuilder as BaseQueryBuilder;
+use RuntimeException;
+
+/**
+ * Defines the command that fixes migrations
+ */
+class FixMigrationsCommand extends Command
+{
+    /** @var string[] The list of migration classes */
+    private $allMigrationClasses = [];
+
+    /** @var string The name of the table to read and write to */
+    protected $tableName = '';
+
+    /** @var IConnection */
+    protected $connection;
+
+    /** @var BaseQueryBuilder */
+    protected $queryBuilder;
+
+    /**
+     * FixMigrationsCommand constructor.
+     *
+     * @param string           $tableName           The name of the table to read and write to
+     * @param string[]         $allMigrationClasses The list of migration classes
+     * @param IConnection      $connection
+     * @param BaseQueryBuilder $queryBuilder
+     */
+    public function __construct(
+        string $tableName,
+        array $allMigrationClasses,
+        IConnection $connection,
+        BaseQueryBuilder $queryBuilder
+    ) {
+        parent::__construct();
+
+        $this->tableName           = $tableName;
+        $this->allMigrationClasses = $allMigrationClasses;
+        $this->connection          = $connection;
+        $this->queryBuilder        = $queryBuilder;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function define()
+    {
+        $this->setName('migrations:fix')
+            ->setDescription('Fixes the migrations database schema');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function doExecute(IResponse $response)
+    {
+        $response->writeln('Fixing migrations database schema...');
+
+        try {
+            $this->connection->beginTransaction();
+
+            switch (get_class($this->queryBuilder)) {
+                case MySqlQueryBuilder::class:
+                    $this->addPrimaryKeyMySql();
+                    break;
+                case PostgreSqlQueryBuilder::class:
+                    $this->addPrimaryKeyPostgreSql();
+                    break;
+                default:
+                    throw new RuntimeException('Unexpected query builder type ' . get_class($this->queryBuilder));
```

Comment:
To make the exception message more helpful, it might be easier to explicitly message the issue, which is `Unsupported database type - only MySQL and PostgreSQL are supported`

Reply:
Fine by me, but it's a copy-paste from here:
https://github.com/opulencephp/Opulence/blob/f04f2b3ca1e20a82fd033cf50f8192d55370cce7/src/Opulence/Framework/Databases/Migrations/SqlExecutedMigrationRepository.php#L147

Should I fix both?

Reply:
No worries - I can fix it after merging